### PR TITLE
feat(tiller): sort resources output by 'helm status'.

### LINKED ISSUE #3127: Order resources in `helm status` output
If you execute `helm status` on the same release, the ordering of the Kubernetes resources changes even if nothing else has.

I'm not sure, but it seems like the ordering is non-deterministic. I believe that they should be ordered deterministically so that multiple calls to `helm status` (or, some people run it inside a `watch` or similar) are easier to read.

### PR DESCRIPTION
**Pull request by silenceshell**
use a slice to store keys of the resources map and sort the slice to print the map later.

Closes #3127

### CODE REVIEW

**Review by bacongobbler — CHANGES_REQUESTED** [review]
Instead of iterating through the keys twice, why don't we write a struct that has comparator functions? That way we can just call `sort.Sort` on the list of structs rather than the key list. It's a little more code, but it prevents us from having to iterate through the range of objects twice.

e.g. https://golang.org/pkg/sort/#example__sortKeys

**Comment by silenceshell** [comment]
Yeah this is much better. I will finish it lately.

**Comment by silenceshell** [comment]
@bacongobbler 
I add a new struct `Obj` for `sort.Sort`, now we just have to iterate once when printing objs. Looking forward to your opinion.

File: pkg/kube/client.go [hunk 1]
```diff
@@ -202,17 +213,26 @@ func (c *Client) Get(namespace string, reader io.Reader) (string, error) {
...
-			if err := p.PrintObj(o, buf); err != nil {
-				c.Log("failed to print object type %s, object: %q :\n %v", t, o, err)
+
+	sort.Sort(Objs(objs))
+
+	var lastVk string
+	for _, t := range objs {
+		vk := t.vk
+		object := t.object
+		if vk != lastVk {
```
Comment (bacongobbler) [inline pkg/kube/client.go]: can you explain this change and how it relates to the old code? This looks like a significant behavioral change here to me from a cursory glance.

Comment (silenceshell) [reply pkg/kube/client.go]: The old codes has two level to store all objs: objs stores different type of objes, and each element in objs(var ot) stores the same type of obj(var o).
The new codes have one flat level to store all objs, and after sorting, all these objs are sorted by their types(v1/Service, v1/Secret, etc), so I need `lastVk` to record the last type, and when type changes, printing a new line and the type header like `==> v1/Service`.

**Comment by bacongobbler** [comment]
Do you think it'd be possible to write a unit test to cover this as well? It'd make myself a little more comfortable to move forward with this change. I understand that this is `pkg/kube` we're talking about though, so please le me know if that's not possible.

Looks like C's failing on this PR too.

**Comment by MPV** [comment]
This would be nice to have when you run `helm` in `watch`. Today it keeps moving around the different resources for each run.

**Comment by bacongobbler** [comment]
closing in favour of #5140

**Pull request closed without merging**